File: sinova-frontend/src/types/preview.ts
import type { DataContext, PreprocessingConfig } from "./preprocessing";

export interface PreviewRequest {
  context: DataContext;
  slice: number;
  configuration: PreprocessingConfig;
  mode: "current" | "original";
}

export type PreviewParams = {
  slice: number;
  configuration: PreprocessingConfig;
  mode?: "current" | "original";
};

export interface PreviewResult {
  context: DataContext;
  width: number;
  height: number;
  data: Float32Array;
  minVal?: number;
  maxVal?: number;
  requestId: string;
  message: string;
}

export interface SinogramViewerProps {
  data?: PreviewResult;
  isLoading: boolean;
  colormap?: string;
}

export interface ProjectionViewerProps {
  data?: PreviewResult;
  isLoading: boolean;
  colormap?: string;
}

export interface ImageStats {
  min: number;
  max: number;
  mean: number;
  stdDev: number;
  snr: number;
}

export interface LineProfileProps {
  data?: PreviewResult;
  isLoading?: boolean;
}

File: sinova-backend/app/schemas/preview.py

from pydantic import BaseModel, Field

from .preprocessing import DataContext, PreprocessingConfiguration


class PreviewRequest(BaseModel):
    context: DataContext
    slice: int = Field(ge=0)
    configuration: PreprocessingConfiguration
    mode: str = "current"


class PreviewResponse(BaseModel):
    context: str
    width: int = Field(gt=0)
    height: int = Field(gt=0)
    data_format: str
    dtype: str
    data: list[float]
    min_value: float | None = None
    max_value: float | None = None
    request_id: str
    message: str